presto-main/src/main/java/io/prestosql/operator/scalar/JoniRegexpFunctions.java

```diff
@@ -307,4 +307,97 @@ private static void validateGroup(long group, Region region)
             throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("Pattern has %d groups. Cannot access group %d", region.numRegs - 1, group));
         }
     }
+
+    private static void validPattern(JoniRegexp pattern)
+    {
+        if (pattern.length() == 0) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "pattern cannot be empty");
+        }
+    }
+
+    @ScalarFunction
+    @Description("returns the beginning position of the matched substring")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.INTEGER)
+    public static long regexpPosition(@SqlType("varchar(x)") Slice source, @SqlType(JoniRegexpType.NAME) JoniRegexp pattern)
+    {
+        return regexpPosition(source, pattern, 1);
+    }
+
+    @ScalarFunction
+    @Description("returns the beginning position of the matched substring from the specified starting position")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.INTEGER)
+    public static long regexpPosition(@SqlType("varchar(x)") Slice source,
+                                      @SqlType(JoniRegexpType.NAME) JoniRegexp pattern,
+                                      @SqlType(StandardTypes.INTEGER) long start)
+    {
+        return regexpPosition(source, pattern, start, 1);
+    }
+
+    @ScalarFunction
+    @Description("returns the beginning position of the nth occurrence substring from the specified starting position")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.INTEGER)
+    public static long regexpPosition(@SqlType("varchar(x)") Slice source,
+                                      @SqlType(JoniRegexpType.NAME) JoniRegexp pattern,
+                                      @SqlType(StandardTypes.INTEGER) long start,
+                                      @SqlType(StandardTypes.INTEGER) long occurrence)
+    {
+        validPattern(pattern);
+        // start should bigger than 0
+        if (start < 1) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "start position cannot small than 1");
+        }
+        // return -1 if start is bigger than the length of source
+        if (start > source.toStringUtf8().length()) {
```

Comment:
You should use `io.airlift.slice.SliceUtf8#countCodePoints(io.airlift.slice.Slice)` instead (this avoids creation of String). However, could we avoid counting of code points here altogether (and rely on `search` to return `-1`)

Reply:
Can't rely on search because we have to convert 'start' from code point to byte point before searching and StringIndexOutOfBoundsException would happen if start is bigger than length. convert code:
```
int nextStart = source.toStringUtf8().substring(0, (int) start - 1).getBytes(StandardCharsets.UTF_8).length;
```